pgrx/src/memcx.rs

```diff
@@ -45,3 +62,54 @@ where
 
     f(&memcx)
 }
+
+#[cfg(all(feature = "nightly", feature = "pg16"))]
+unsafe impl<'mcx> std::alloc::Allocator for &MemCx<'mcx> {
+    fn allocate(
+        &self,
+        layout: std::alloc::Layout,
+    ) -> Result<NonNull<[u8]>, std::alloc::AllocError> {
+        unsafe {
+            // Bitflags for MemoryContextAllocAligned:
+            // #define MCXT_ALLOC_HUGE    0x01 /* allow huge allocation (> 1 GB) */
+            // #define MCXT_ALLOC_NO_OOM  0x02 /* no failure if out-of-memory */
+            // #define MCXT_ALLOC_ZERO    0x04 /* zero allocated memory */
+            let ptr = pgrx_pg_sys::MemoryContextAllocAligned(
+                self.ptr.as_ptr(),
+                layout.size(),
+                layout.align(),
+                0,
+            );
+            let slice: &mut [u8] = slice::from_raw_parts_mut(ptr.cast(), layout.size());
+            Ok(NonNull::new_unchecked(slice))
+        }
+    }
+
+    unsafe fn deallocate(&self, ptr: NonNull<u8>, _layout: std::alloc::Layout) {
+        // TODO: Find faster free for use when MemoryContext is known.
+        // This is the global function that looks up the relevant Memory Context by address range.
+        pgrx_pg_sys::pfree(ptr.as_ptr().cast())
```

Comment:
They do this via a macro that accesses a `static const` and calls a `static inline` function, so we have no realistic way of doing better without PRing Postgres, basically.

nit:
```suggestion
        pg_sys::pfree(ptr.as_ptr().cast())
```